### ISSUE TITLE
Port embedding API to plugin

### ISSUE THREAD

**Issue by AleksKnezevic** [tenstorrent/tt-xla#1366]
We need to port embedding API support in the vLLM plugin that will be part of this repo #1365.

**Comment by mmanzoorTT** [comment]
vLLM Embedding API update
- Pre-compilation of multiple binaries for different input sequence lengths.
- Process multiple batches; A batch contains one or more input sequences. 
- Next step is to include `attention_mask`; currently, we are relying on tt-metal default `attention_mask`

**Comment by AleksKnezevic** [comment]
@LPanosTT added that part, so he can better comment.

**Comment by LPanosTT** [comment]
@bgoelTT The inputs are padded to the next highest power of 2 when vllm receives them and it runs that pre-compiled version of the model. The user entering prompts doesn't need to pad anything themselves.

**Comment by LPanosTT** [comment]
@bgoelTT when vllm receives the request it pads them automatically to the next power of 2 before executing the model.

**Comment by mmanzoorTT** [comment]
A little bit more detail: `execute_model` function receives `SchedulerOutput` object which contains the unpadded input tokens as `list` (This input tokens contain one or more input sequences). `_update_states` function create/add requests for these input sequence(s) and then `_prepare_inputs` function generates padded input tokens which is then fed to the model in `execute_model` function. `max_num_seqs` flag is used to control the max number of sequences in a batch. 

For example (max_num_seqs=3 to process three input prompts in a batch), If we have 3 prompts of length 10, 12, and 14 then vllm will give us unpadded input (list of length 36) and we create a tensor of [1 x 64] (including padding)

**Comment by AleksKnezevic** [comment]
@mmanzoorTT, can we close?
